hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/logs/container/utils/ContainerDatanodeDatabase.java

```diff
@@ -0,0 +1,555 @@
+
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.hadoop.hdds.conf.OzoneConfiguration;
+import org.apache.hadoop.hdds.protocol.proto.HddsProtos;
+import org.apache.hadoop.hdds.scm.container.ReplicationManagerReport;
+import org.apache.hadoop.ozone.om.OMConfigKeys;
+import org.sqlite.SQLiteConfig;
+
+/**
+ * Handles creation and interaction with the database.
+ * Provides methods for table creation, log data insertion, and index setup.
+ */
+public class ContainerDatanodeDatabase {
+  
+  private static String databasePath;
+  private static final int DEFAULT_REPLICATION_FACTOR;
+  
+  static {
+    OzoneConfiguration configuration = new OzoneConfiguration();
+    final String replication = configuration.getTrimmed(
+        OMConfigKeys.OZONE_SERVER_DEFAULT_REPLICATION_KEY,
+        OMConfigKeys.OZONE_SERVER_DEFAULT_REPLICATION_DEFAULT);
+
+    DEFAULT_REPLICATION_FACTOR = Integer.parseInt(replication.toUpperCase());
+  }
+  
+  public static void setDatabasePath(String dbPath) {
+    if (databasePath == null) {
+      databasePath = dbPath;
+    }
+  }
+
+  private static Connection getConnection() throws Exception {
+    if (databasePath == null) {
+      throw new IllegalStateException("Database path not set");
+    }
+    
+    Class.forName(SQLDBConstants.DRIVER);
+
+    SQLiteConfig config = new SQLiteConfig();
+
+    config.setJournalMode(SQLiteConfig.JournalMode.OFF);
+    config.setCacheSize(SQLDBConstants.CACHE_SIZE);
+    config.setLockingMode(SQLiteConfig.LockingMode.EXCLUSIVE);
+    config.setSynchronous(SQLiteConfig.SynchronousMode.OFF);
+    config.setTempStore(SQLiteConfig.TempStore.MEMORY);
+
+    return DriverManager.getConnection(SQLDBConstants.CONNECTION_PREFIX + databasePath, config.toProperties());
+  }
+
+  public void createDatanodeContainerLogTable() throws SQLException {
+    String createTableSQL = SQLDBConstants.CREATE_DATANODE_CONTAINER_LOG_TABLE;
+    try (Connection connection = getConnection();
+         Statement dropStmt = connection.createStatement();
+         Statement createStmt = connection.createStatement()) {
+      dropTable(SQLDBConstants.DATANODE_CONTAINER_LOG_TABLE_NAME, dropStmt);
+      createStmt.execute(createTableSQL);
+      createDatanodeContainerIndex(createStmt);
+    } catch (SQLException e) {
+      System.err.println("Error while creating the table: " + e.getMessage());
+      throw e;
+    } catch (Exception e) {
+      System.err.println("Unexpected error: " + e.getMessage());
+      throw new RuntimeException(e);
```

Comment:
Same as above,
```suggestion
      throw new RuntimeException(""Unexpected error: "  + e);
```
We can do the same for the other occurrences in the file.